plugins/in_windows_exporter_metrics/we_wmi.c

```diff
@@ -216,32 +212,41 @@ static inline int wmi_update_metrics(struct flb_we *ctx, struct wmi_query_spec *
     VARIANT prop;
     int label_index = 0;
     HRESULT hr;
-    char *strlabel;
+    char *strlabel = NULL;
     char *metric_label_set[WE_WMI_METRIC_LABEL_LIST_SIZE];
     int metric_label_count = 0;
     char buf[16] = {0};
     wchar_t *wlabel;
+    char *newstr = NULL;
+    VARIANT dstProp;
 
     VariantInit(&prop);
+    VariantInit(&dstProp);
     metric_label_count = 0;
     for (label_index = 0; label_index < spec->label_property_count; label_index++) {
         wlabel = we_convert_str(spec->label_property_keys[label_index]);
         hr = class_obj->lpVtbl->Get(class_obj, wlabel, 0, &prop, 0, 0);
-        switch(prop.vt) {
-        case VT_I4:
-            snprintf(buf, 16, "%d", prop.lVal);
-            metric_label_set[label_index] = strdup(buf);
-            metric_label_count++;
-            break;
-        case VT_BSTR:
-            strlabel = we_convert_wstr(prop.bstrVal, CP_UTF8);
-            metric_label_set[label_index] = strdup(strlabel);
-            metric_label_count++;
+        if (prop.vt == VT_NULL) {
```

Comment:
Please take this conversion logic, extract it in a small static function that accepts a variant value pointer and flag to determine if `VT_NULL` should be turned into an empty string or just return `NULL` and then use that one function in the three parts of the code where you are currently doing this.

Reply:
I extracted a static function that uses variant pointer and flag arguments: https://github.com/fluent/fluent-bit/pull/7395/commits/b74708032cafea40ea4e0fdb62fb53f43b40a233